packages/backend-app-api/package.json

```diff
@@ -90,6 +90,7 @@
     "@types/node-forge": "^1.3.0",
     "@types/stoppable": "^1.1.0",
     "http-errors": "^2.0.0",
+    "msw": "^1.0.0",
```

Comment:
MSW v2 is nice but its API is completely different than the API of MSW v1.
The migration is not straight forward. There is a [codemod](https://mswjs.io/docs/migrations/1.x-to-2.x#codemods) to help with the migration but it covers only about 80% of the required changes. Adding new tests that use MSW v1 require more work to migrate them to v2 later.

We tried to migrate the tests to MSW v2 in our application and we encountered the first issue described in the [MSW v1 to v2 migration page](https://mswjs.io/docs/migrations/1.x-to-2.x) ([`TextEncoder` is not defined (Jest)](https://mswjs.io/docs/migrations/1.x-to-2.x#requestresponsetextencoder-is-not-defined-jest).)
The workaround presented there does not work in a Backstage application because there isn't a global `jest.config.js` file where to configure the provided polyfill.

Is there a way to implement this workaround in a single place in our Backstage-based application?
Or should we wait for Backstage to migrate to MSW v2? (If you put the fix in the Jest configuration of Backstage, I hope that it will fix the issue for the applications, too.)

Reply:
Yes, you [put them in your `setupTests.ts` file](https://github.com/backstage/backstage/blob/96b23786f648ea28c2ab81d4888c46ab8bbd907b/plugins/api-docs/src/setupTests.ts#L19).

I don't mind very much that there's some msw1 based tests, not everything needs to go through the migration pain. But yeah I think it's time that newly written things start considering msw2 at least.

Reply:
Thank you for your answer. We can put it in `setupTests.ts`, in the affected plugins only, but having it in a common configuration file prevents the problem occur in the plugins that we write in the future, without requiring additional setup.

Indeed, putting the polyfill in the configuration provided by Backstage is probably not the best approach. However, I think that Backstage should provide a way for the applications to have a global configuration file for Jest (a `jest.config.js` file in the root directory of the project that contains settings common for all plugins/packages), allowing each plugin or package to override parts of this global configuration, if they need, via `setupTests.ts` or by other means (`jest.config.js`).

If I am not wrong, now the `jest.config.js` files located in the plugins are read but a global `jest.config.js` file is ignored.